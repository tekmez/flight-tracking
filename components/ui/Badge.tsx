import { View, Text } from "react-native";
import React from "react";

const Badge = () => {
  return (
    <View className="py-1 px-[10px] border-[0.5px] rounded-[7px] border-orange-200 bg-orange-50">
      <Text className="font-sfProMedium text-[12px] leading-4 text-orange-700 text-center">
        9h 50m
      </Text>
    </View>
  );
};

export default Badge;
