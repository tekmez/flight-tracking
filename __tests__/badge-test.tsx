import Badge from "@/components/ui/Badge";
import { render } from "@testing-library/react-native";

describe("Badge", () => {
  it("renders correctly", () => {
    const tree = render(<Badge duration="1000" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders correctly with duration", () => {
    const { getByText } = render(<Badge duration="1000" />);
    const badgeText = getByText("1000");
    expect(badgeText.props.children).toEqual("1000");
  });
});
