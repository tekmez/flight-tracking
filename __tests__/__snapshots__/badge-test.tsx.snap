// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders correctly 1`] = `
<View
  style={
    [
      {
        "paddingBottom": 4,
        "paddingTop": 4,
      },
      {
        "paddingLeft": 10,
        "paddingRight": 10,
      },
      {
        "borderBottomWidth": 0.5,
        "borderLeftWidth": 0.5,
        "borderRightWidth": 0.5,
        "borderTopWidth": 0.5,
      },
      {
        "borderBottomLeftRadius": 7,
        "borderBottomRightRadius": 7,
        "borderTopLeftRadius": 7,
        "borderTopRightRadius": 7,
      },
      {
        "borderBottomColor": "#fed7aa",
        "borderLeftColor": "#fed7aa",
        "borderRightColor": "#fed7aa",
        "borderTopColor": "#fed7aa",
      },
      {
        "backgroundColor": "#fff7ed",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "SFProMedium",
        },
        {
          "fontSize": 12,
        },
        {
          "lineHeight": 16,
        },
        {
          "color": "#c2410c",
        },
        {
          "textAlign": "center",
        },
      ]
    }
  >
    1000
  </Text>
</View>
`;
